sqswatcher/sqswatcher.py

```diff
@@ -185,10 +193,21 @@ def pollQueue(scheduler, q, t, proxy_config):
                             break
 
                     log.info("instanceId=%s" % instanceId)
+                    log.info("Fetching the item from DynamoDB table")
                     try:
                         item = t.get_item(ConsistentRead=True, Key={"instanceId": instanceId})
                         if item.get('Item') is not None:
                             hostname = item.get('Item').get('hostname')
+                            terminated_by_cluster = item.get('Item').get('terminated_by_cluster')
+
+                            if not terminated_by_cluster:
+                                log.info("Instance %s was not terminated by the cluster" % instanceId)
+                                t.update_item(
```

Comment:
What about creating a new class to manage the db and move all the table management functions on it?